source-container-build/app/source_build.py

```diff
@@ -445,6 +445,19 @@ def resolve_source_image_by_manifest(image: str) -> str | None:
         log.info("Source container image %s does not exist.", source_image)
 
 
+def resolve_source_image(binary_image: str, registries_allow_list: list[str]) -> str | None:
+    allowed = urlparse("docker://" + binary_image).netloc in registries_allow_list
+    if allowed:
+        return resolve_source_image_by_version_release(binary_image)
+    else:
+        logger.info(
+            "Image %s does not come from supported allowed registry. "
+            "Skip handling the sources for it.",
+            binary_image,
+        )
+    return resolve_source_image_by_manifest(binary_image)
```

Comment:
Even if the image is in the "allowlist" (more like a list of registries where we expect to find `{version}-{release}.src` tags), we should still `resolve_source_image_by_manifest` for it.

Example: not all the images in `registry.access.redhat.com` use the `{version}-{release}.src` tagging scheme, some are built and released by Konflux and use the "cosign-style" tagging.

Reply:
I know we discussed this offline, but can't quite remember what we agreed on. Could you summarize it here?